cytoolz/itertoolz.pyx

```diff
@@ -1041,7 +1044,8 @@ cdef class partition_all:
     def __next__(self):
         cdef tuple result
         cdef object item
-        cdef Py_ssize_t i = 0
+        cdef object none = None
+        cdef Py_ssize_t i = 0, j
```

Comment:
mixing initialization and deceleration is confusing imho, you could just split this to two lines.